### ISSUE TITLE
Otto dev generates invalid IP address

### ISSUE THREAD

**Issue by jtcwang** [hashicorp/otto#412]
It seems like when running `Otto dev` dev it will generate bad IP like `100.107.92.255`

My friend ran through the `otto-getting-started` example and got the IP above. He wasn't able to connect to `100.107.92.255:9292`.

running `ip a` shows this.

```
eth1: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc pfifo_fast state UP qlen 1000
    link/ether 08:00:27:9b:79:0e brd ff:ff:ff:ff:ff:ff
    inet 100.107.92.255/24 brd 100.107.92.255 scope global eth1
    inet6 fe80::a00:27ff:fe9b:790e/64 scope link 
       valid_lft forever preferred_lft forever
```

We changed the address using the following command:
```
sudo ip addr del 100.107.92.255/24 dev eth1
sudo ip addr add 100.107.92.254/24 dev eth1
```
and connecting to `100.107.92.254:9292` now works.
Apparently 255 is reserved (for broadcast) but my knowledge is very poor in networking. Can someone double check how the IP are generated and make sure it account for this?

**Comment by n00bsys0p** [comment]
This is also occurring for me. I can connect to the IP fine, but it is not within a private CIDR range. I note that there does seem to be some code relating to the usual private ranges [here](https://github.com/hashicorp/otto/blob/0d11db6a913cbdee5b584187f57a28bb7e264eff/helper/localaddr/subnet.go).

However the IP address that it's assigned to my dev box is 100.74.232.194, which is not under [RFC1989 ranges](https://tools.ietf.org/html/rfc1918) 172.16.0.0/12, 10.0.0.0/8 or 192.168.0.0/16, or even APIPA (which would be an odd choice, admittedly). I've done a little investigation into this, and it appears that this range is a fairly new and small private address range as specified in [RFC6598](https://tools.ietf.org/html/rfc6598), which covers the range 100.64.0.0/10.

Could I ask for a little detail on why you chose such a range, considering the enormous number of available private subnets, and the relatively trivial work that has to be done to check that none are in use on the local machine? Is it simply to avoid any potential issues with complex or unexpected local routing on development machines or internal corporate development networks?

I'm very interested to understand the motivation behind this, particularly if it's technologically influenced.

**Comment by jtcwang** [comment]
I saw your old post and I think it's relevant as well?

**Comment by n00bsys0p** [comment]
Hey - my comment wasn't relevant, because [the 100.64.0.0/10 IP range is a valid private range](https://tools.ietf.org/html/rfc6598) - the problem you're seeing is you were for some reason assigned a broadcast address (255). I reposted my own question [as an issue of its own](https://github.com/hashicorp/otto/issues/413) as I'm interested to learn why they chose an RFC6598 address range instead of [one of the "normal" RFC1918 ranges](https://tools.ietf.org/html/rfc1918).

**Comment by leowmjw** [comment]
@jtcwang FYI; since the CIDR is 100.64.0.0/10; the netmask in the Vagrantfile needs to be set to match; in that case the IP address you were allocated with would have worked as broadcast address would have been 100.127.255.255

**Comment by jtcwang** [comment]
I'm not familiar with networking in general, but thanks for the info. Perhaps it's time for me to read up on some of these things :)

Feel free to close whenever.

**Comment by n00bsys0p** [comment]
@leowmjw Ah that's what I missed! I didn't notice the 24 bit netmask!

**Comment by bigwheel** [comment]
Im meeting same problem maybe.

```
eth1      Link encap:Ethernet  HWaddr 08:00:27:6e:01:9d  
          inet addr:100.122.188.0  Bcast:100.122.188.255  Mask:255.255.255.0
          inet6 addr: fe80::a00:27ff:fe6e:19d/64 Scope:Link
          UP BROADCAST RUNNING MULTICAST  MTU:1500  Metric:1
          RX packets:9 errors:0 dropped:0 overruns:0 frame:0
          TX packets:6 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:1000 
          RX bytes:1452 (1.4 KB)  TX bytes:468 (468.0 B)
```

I saw this odd Ip Address in [Otto Getting Started in Official Document](https://www.ottoproject.io/intro/getting-started/dev.html).
Otto version is v0.2.0

**Comment by leowmjw** [comment]
@bigwheel Yup; looks like it hit the Network Address instead this time.  You can apply @jtcwang's method to fix it.  My have my PR which will fix this; pending for Hashicorp folks approval.

**Comment by bigwheel** [comment]
ty leowmjw
